File: frontend/src/api/review-meeting.ts
import api from './index'

export interface ReviewMeetingItem {
  id?: number
  projectId: number
  stageId?: number
  meetingCode: string
  meetingName: string
  meetingType: string
  meetingDate: string
  meetingLocation: string
  hostUserId?: number
  attendeeUsers?: string
  expertGroup?: string
  status: string
}

export function getReviewMeetings(params: {
  pageNo?: number
  pageSize?: number
  projectId?: number
  stageId?: number
  status?: string
}) {
  return api.get('/review-meetings', { params })
}

export function getReviewMeeting(id: number) {
  return api.get(`/review-meetings/${id}`)
}

export function createReviewMeeting(data: ReviewMeetingItem) {
  return api.post('/review-meetings', data)
}

export function updateReviewMeeting(id: number, data: ReviewMeetingItem) {
  return api.put(`/review-meetings/${id}`, data)
}

export function updateMeetingStatus(id: number, status: string) {
  return api.put(`/review-meetings/${id}/status`, null, { params: { status } })
}

export function deleteReviewMeeting(id: number) {
  return api.delete(`/review-meetings/${id}`)
}

File: backend/src/main/java/com/military/doc/modules/review/entity/ReviewMeeting.java
package com.military.doc.modules.review.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("review_meeting")
public class ReviewMeeting {
    @TableId(type = IdType.AUTO)
    private Long id;

    private Long projectId;
    private Long stageId;
    private String meetingCode;
    private String meetingName;
    private String meetingType;
    private LocalDate meetingDate;
    private String meetingLocation;
    private Long hostUserId;
    private String attendeeUsers;
    private String expertGroup;
    private String status;

    @TableField(fill = FieldFill.INSERT)
    private Long createdBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updatedBy;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableLogic
    private Integer deleted;
}